driver/razeraccessory_driver.c

```diff
@@ -481,6 +500,40 @@ static ssize_t razer_attr_write_mode_static(struct device *dev, struct device_at
             report.transaction_id.id = 0x1F;
             break;
 
+        case USB_DEVICE_ID_RAZER_CHARGING_PAD_CHROMA:
+            // Must be in normal mode for hardware effects
+            razer_set_device_mode(device->usb_dev, 0x00, 0x00);
+            /**
+             * Mode switcher required after setting static color effect once and before setting a second time.
+             * Similar to Naga Trinity?
+             *
+             * If the color is not set twice with the mode switch inbetween, each subsequent
+             * setting of the static effect actually sets the previous color...
+             */
+            report = razer_chroma_extended_matrix_effect_static(VARSTORE, ZERO_LED, (struct razer_rgb*) & buf[0]);
+            report.transaction_id.id = 0x1F;
+
+            mutex_lock(&device->lock);
+            razer_send_payload(device->usb_dev, &report);
+            mutex_unlock(&device->lock);
+
+            report = get_razer_report(0x0f, 0x02, 0x06);
+            report.arguments[0] = 0x00;
+            report.arguments[1] = 0x00;
+            report.arguments[2] = 0x08;
+            report.arguments[3] = 0x00;
+            report.arguments[4] = 0x00;
+            report.arguments[5] = 0x00;
+            report.transaction_id.id = 0x1F;
```

Comment:
Any idea what this kind of does? Two new Naga mice also use this..

Reply:
In my testing if I sent just that command, it's as if the device were told to disable the current effect. The lighting just goes dark. And I was seeing that in the transition of static effect pcaps. I just so happened to remember the openrazer wiki had a list of unknown commands. So I went and found reference to that command with the Naga Trinity mentioned. I think it was also mentioned in #541

It seems to disable the current effect, but also prepares the device so it can accept the next static effect command? I'm not sure why it's needed or what it's doing exactly. I wouldn't know how or if it is even possible to reverse engineer the command any further. It is very strange.

Reply:
OH, HERE'S AN IDEA! Maybe a way to set a custom frame on this device while not requiring a mode switch?

Reply:
@z3ntu If you have any suggestions on how we might go about digging up more info on this command, or if you would like me to test anything, I'm absolutely willing.

Reply:
As I'm not aware that you can generate these packets with Synapse on their own, I think it'll stay a mystery :)